api/v1alpha4/cluster_types.go

```diff
@@ -60,6 +60,64 @@ type ClusterSpec struct {
 	// for provisioning infrastructure for a cluster in said provider.
 	// +optional
 	InfrastructureRef *corev1.ObjectReference `json:"infrastructureRef,omitempty"`
+
+	// This encapsulates the topology for the cluster.
+	// +optional
+	Topology *Topology `json:"topology,omitempty"`
+}
+
+// Topology encapsulates the information of the managed resources.
+type Topology struct {
+	// The name of the ClusterClass object to create the topology.
+	Class string `json:"class,omitempty"`
+
+	// The kubernetes version of the cluster.
+	Version string `json:"version"`
+
+	// RolloutAfter performs a rollout of the entire cluster one component at a time [...]
+	// +optional
+	RolloutAfter *metav1.Time `json:"rolloutAfter,omitempty"`
+
+	// The information for the Control plane of the cluster.
+	ControlPlane ControlPlane `json:"controlPlane"`
+
+	// Worker encapsulates the different constructs that form the worker nodes
+	// for the cluster.
+	// +optional
+	Worker *Worker `json:"worker,omitempty"`
+}
+
+// ManagedMachineDeployment specifies the different parameters for a set of worker nodes in the topology.
+// This set of nodes is managed by a MachineDeployment object whose lifecycle is managed by the Cluster controller.
+type ManagedMachineDeployment struct {
+	Metadata ObjectMeta `json:"metadata,omitempty"`
+
+	// Class is the name of the DeploymentClass used to create the set of worker nodes.
+	// This should match one of the deployment classes defined in the ClusterClass object
+	// mentioned in the `Cluster.Spec.Class` field.
+	Class string `json:"class"`
+
+	// Name refers to the name of the set of worker nodes.
+	Name string `json:"name"`
```

Comment:
Should we mention that the name must be unique within the list of managed workers?

Reply:
Done.
I did the same for class within ClusterClass.Workers.Deployments